lisa/notifiers/junit.py

```diff
@@ -302,18 +310,27 @@ def _add_test_case_result(
 
         if message.status == TestStatus.FAILED:
             failure = ET.SubElement(testcase, "failure")
-            failure.attrib["message"] = message.message
-            failure.text = message.stacktrace
-
+            if runbook.include_failed_messages:
+                failure.attrib["message"] = message.message
+                # failure.text = message.stacktrace
+                if str(message.stacktrace) != "":
+                    failure.attrib["stacktrace"] = str(message.stacktrace)
```

Comment:
Is this defined by junit schema?

Reply:
This is compliant with Jnuit schema.  
Its expected to have stacktrace even for successful cases. in our case, we include the  <case>.dmesg output from results folder here